logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationListDetailsUseCaseImpl.kt

```diff
@@ -21,32 +22,47 @@ internal class ObserveConversationListDetailsUseCaseImpl(
     private val callRepository: CallRepository,
 ) : ObserveConversationListDetailsUseCase {
 
-    override suspend operator fun invoke(): Flow<List<ConversationDetails>> {
+    override suspend operator fun invoke(): Flow<ConversationListDetails> {
         syncManager.startSyncIfIdle()
 
-        val conversationsFlow = conversationRepository.observeConversationList().map { conversations ->
-            conversations.map { conversation ->
-                flow {
-                    emit(null)
-                    emitAll(conversationRepository.observeConversationDetailsById(conversation.id))
-                }
-            }
-        }.flatMapLatest { flowsOfDetails ->
-            combine(flowsOfDetails) { latestValues -> latestValues.asList().mapNotNull { it } }
-        }
-
-        return combine(conversationsFlow, callRepository.ongoingCallsFlow()) { conversations, calls ->
+        return combine(observeLatestConversationDetails(), callRepository.ongoingCallsFlow()) { conversations, calls ->
             conversations.map {
                 when (it) {
                     is ConversationDetails.Self,
                     is ConversationDetails.Connection,
                     is ConversationDetails.OneOne -> it
-
                     is ConversationDetails.Group -> it.copy(
                         hasOngoingCall = (it.conversation.id in calls.map { call -> call.conversationId })
                     )
                 }
             }
+        }.map { conversationList ->
+            ConversationListDetails(
+                conversationList = conversationList,
+                //TODO: any better idea what to do on StorageFailure ?
+                unreadConversationsCount = conversationRepository.getUnreadConversationCount().fold({ 0 }, { it })
+            )
+        }
+    }
+
+    private suspend fun observeLatestConversationDetails(): Flow<List<ConversationDetails>> {
+        return conversationRepository.observeConversationList().map { conversations ->
+            conversations.map { conversation ->
+                flow {
+                    emit(null)
+                    emitAll(conversationRepository.observeConversationDetailsById(conversation.id))
+                }
+            }
+        }.flatMapLatest { flowsOfDetails ->
+            combine(flowsOfDetails) { latestValues -> latestValues.asList().mapNotNull { it } }
```

Comment:
interesting solution! maybe extract it to some extended fun (`com.wire.kalium.logic.functional.Flow` is a good place for it)

so it would be like 
```
private suspend fun observeLatestConversationDetails(): Flow<List<ConversationDetails>> = 
   conversationRepository.observeConversationList()
      .flatMapEachItem { conversation ->  conversationRepository.observeConversationDetailsById(conversation.id) } // the name could be better :) 
```

Reply:
thanks :P but i was not the done writing it xd but will move it there

Reply:
I hope you do not mind, but I'll leave as it is